frontend/class-frontend.php

```diff
@@ -988,7 +988,7 @@ private function generate_canonical() {
 		 *
 		 * @api string $canonical The canonical URL
 		 */
-		$this->canonical = apply_filters( 'wpseo_canonical', $canonical );
+		$this->canonical = apply_filters( 'wpseo_canonical', user_trailingslashit( $canonical ) );
```

Comment:
When pagination is done with Query String arguments this will give unwanted results.